lightly/data/lighty_subset.py

```diff
@@ -0,0 +1,37 @@
+from typing import List, Dict, Tuple
+
+from lightly.data.dataset import LightlyDataset
+
+
+class LightlySubset(LightlyDataset):
+    def __init__(self, base_dataset: LightlyDataset, filenames_subset: List[str]):
+        self.base_dataset = base_dataset
+
+        dict_base_dataset_filename_index: Dict[str, int] = \
+            dict([
+                (base_dataset.index_to_filename(self.dataset, index), index)
+                for index in range(len(base_dataset))
+            ])
+        self.mapping_subset_index_baseset = \
+            [dict_base_dataset_filename_index[filename] for filename in filenames_subset]
+
+    def __getitem__(self, index_subset: int) -> Tuple[object, object, str]:
+        index_baseset = self.mapping_subset_index_baseset[index_subset]
+        sample, target, fname = self.base_dataset.__getitem__(index_baseset)
```

Comment:
You could also leave away this extra line but that's just a minor thing.

Reply:
I prefer naming the return values explicitly, as it makes it much easier and faster during coding to assign the return values of a call of this function to variables:
```python
for i in range(len(dataset)):
    sample, target, fname = dataset.__getitem__(i)
```
Without naming the variables explicitly, I would need to dig into the nested function `self.base_dataset.__getitem__(index_baseset)` and possibly even further to get what the return values are and stand for.